src/lib/InstantSearch.ts

```diff
@@ -245,23 +246,22 @@ See: https://www.algolia.com/doc/guides/building-search-ui/going-further/backend
   }
 
   /**
-   * Adds a widget. This can be done before and after InstantSearch has been started. Adding a
-   * widget after InstantSearch started is considered **EXPERIMENTAL** and therefore
-   * it is possibly buggy, if you find anything please
-   * [open an issue](https://github.com/algolia/instantsearch.js/issues/new?title=Problem%20with%20hot%20addWidget).
-   * @param  widget The widget to add to InstantSearch. Widgets are simple objects
-   * that have methods that map the search life cycle in a UI perspective. Usually widgets are
-   * created by [widget factories](widgets.html) like the one provided with InstantSearch.js.
+   * Adds a widget. This can be done before and after InstantSearch has been started.
+   * @param widget The widget to add to InstantSearch.
+   *
+   * This method is deprecated, and `addWidget(widget)` can be replaced by `addWidgets([widget])`
    */
   public addWidget(widget: Widget) {
+    warning(
+      false,
+      'addWidget(widget) is replaced by addWidgets([widgets]) to help see the structure'
+    );
     this.addWidgets([widget]);
   }
 
   /**
-   * Adds multiple widgets. This can be done before and after the InstantSearch has been started. This feature
-   * is considered **EXPERIMENTAL** and therefore it is possibly buggy, if you find anything please
-   * [open an issue](https://github.com/algolia/instantsearch.js/issues/new?title=Problem%20with%20addWidgets).
-   * @param {Widget[]} widgets The array of widgets to add to InstantSearch.
+   * Adds multiple widgets. This can be done before and after the InstantSearch has been started.
+   * @param widgets The array of widgets to add to InstantSearch.
    */
   public addWidgets(widgets: Widget[]) {
```

Comment:
What about this syntax?

```ts
// Signature
addWidgets(...widgets: Widget[])

// Usage
search.addWidgets(searchBox, hits, pagination)
```

Reply:
Interesting idea, but i'm not sure how to not make that a big breaking change for anyone now

Reply:
Maybe open an issue about it

Reply:
Agee with @Haroenv. The idea with 4.x.x is to ease the transition as much as possible. If it was the introduction of the function I would have been for it, but it's not the case. The function is already there for quite some time now. I would prefer stick the current version.

Reply:
Yep, just wanted to share this idea. We don't document the array usage as much as the `addWidget` so I figured I'd trigger the discussion.